### ISSUE TITLE
1.0rc1: many ascii read failures on Debian Hurd

### ISSUE THREAD

**Issue by olebole** [astropy/astropy#3416]
On Debian hurd, semaphores are no implemented, which causes lots of errors in `astropy/io/ascii/tests/test_c_reader.py`. While the platform is quite exotic, it may be relatively easy to fix; all failures trace back to `astropy/io/ascii/ui.py:158`:
```
        # Try the fast reader first if applicable
        if fast_reader_param and format is not None and 'fast_{0}'.format(format) \
                                                        in core.FAST_CLASSES:
            new_kwargs['Reader'] = core.FAST_CLASSES['fast_{0}'.format(format)]
            fast_reader = get_reader(**new_kwargs)
            try:
                return fast_reader.read(table)
            except (core.ParameterError, cparser.CParserError) as e:
                # special testing value to avoid falling back on the slow reader
                if fast_reader_param == 'force':
                    raise e
                # If the fast reader doesn't work, try the slow version
                dat = reader.read(table)
        else:
            dat = reader.read(table)
```
Full log [here](https://buildd.debian.org/status/fetch.php?pkg=python-astropy&arch=hurd-i386&ver=1.0~rc1-1&stamp=1422724845).

Adding `ImportError` to the error list above fixes these exceptions, however it may be better to catch the ImportError already in `astropy/io/ascii/cparser.pyx:392` (and raise a `cparser.CParserError` instead).

**Comment by olebole** [comment]
I created a pull request in #3417

**Comment by olebole** [comment]
Sorry, please re-open this. On some python versions (namely 3.4.2), the exception raised is not an ImportError but a NotImplementedException. In my last tests this was hidden by other errors. I will prepare a new pull request (or shall I just update #3425 ?)

**Comment by astrofrog** [comment]
@olebole - since #3425 is already merged you will need to open a new pull request. Thanks!

**Comment by olebole** [comment]
Sorry that I request reopening this again :-( However, a test compilation of an older kFreeBSD (kernel version 9) showed that there is another type of exception to be catched: `OSError`. Patch is trivial, however not urgent (since the old version is not my primary target). I'll prepare one...

**Comment by astrofrog** [comment]
(Closed again since #3509 is merged)
